fix(ci): e2e-ci fails on every PR — @langwatch/mcp-server dist/ not built before app startup

## Problem

The `sdk-javascript-ci / e2e` and `e2e-ci` jobs fail on both PRs and main with:

```
Error: Cannot find module '/home/runner/work/langwatch/langwatch/langwatch/node_modules/@langwatch/mcp-server/dist/create-mcp-server.js'
```

The app server (`langwatch/src/server.ts`) imports from `@langwatch/mcp-server`, but CI workflows don't build the MCP server package before starting the app. The `dist/` directory doesn't exist at runtime.

After this crash, the app never comes up on port 5570 and e2e global-setup times out after 30 fetch retries.

## Evidence

- Failing on main: https://github.com/langwatch/langwatch/actions/runs/24191651059
- Same failure on PR branches — not caused by any PR changes
- Two `e2e-ci` workflow runs trigger on every PR push (duplicate workflow files); one passes, the other fails

## Root causes

1. **Missing build step** — `@langwatch/mcp-server` is a workspace package whose `dist/` must be built before `pnpm start:app`. Neither `sdk-javascript-ci.yml` nor `e2e-ci` workflows build it.
2. **Duplicate e2e-ci workflows** — two workflow files match, causing duplicate runs that race on the same build artifacts.

## Fix

- Add a build step for `mcp-server` before app startup in CI:
  ```yaml
  - name: Build MCP server
    run: cd mcp-server && pnpm build
  ```
- Identify and remove the duplicate `e2e-ci` workflow definition
- Optionally: add a sanity check that verifies workspace dist dirs exist before starting the app

## Impact

Blocks PRs from going fully green even when the code is correct. Requires maintainer intuition to distinguish real failures from this infra issue.

--- a/.github/workflows/e2e-ci.yml
+++ b/.github/workflows/e2e-ci.yml
@@ -114,6 +114,7 @@ jobs:
           cache-dependency-path: |
             langwatch/pnpm-lock.yaml
             agentic-e2e-tests/pnpm-lock.yaml
+            mcp-server/pnpm-lock.yaml
 
       - name: Install app dependencies
         working-directory: langwatch
@@ -151,6 +152,10 @@ jobs:
       - name: Setup Elasticsearch
         working-directory: langwatch
         run: pnpm elastic:migrate
+
+      - name: Build mcp-server
+        working-directory: mcp-server
+        run: pnpm install --frozen-lockfile && pnpm run build
 
       - name: Build app
         working-directory: langwatch

--- a/.github/workflows/e2e-ci.yml
+++ b/.github/workflows/e2e-ci.yml
@@ -162,6 +162,11 @@ jobs:
         run: NODE_ENV=test pnpm build
 
       - name: Run E2E tests
+        env:
+          # Migrations already ran in earlier steps; skip them during pnpm start
+          SKIP_PRISMA_MIGRATE: "true"
+          SKIP_ELASTIC_MIGRATE: "true"
+          SKIP_CLICKHOUSE_MIGRATE: "true"
         run: |
           # Start the app in background
           cd langwatch && PORT=5570 pnpm start &

--- a/.github/workflows/e2e-ci.yml
+++ b/.github/workflows/e2e-ci.yml
@@ -116,6 +116,10 @@ jobs:
             agentic-e2e-tests/pnpm-lock.yaml
             mcp-server/pnpm-lock.yaml
 
+      - name: Build mcp-server
+        working-directory: mcp-server
+        run: pnpm install --frozen-lockfile && pnpm run build
+
       - name: Install app dependencies
         working-directory: langwatch
         run: CI=true pnpm install --frozen-lockfile
@@ -152,10 +156,6 @@ jobs:
       - name: Setup Elasticsearch
         working-directory: langwatch
         run: pnpm elastic:migrate
-
-      - name: Build mcp-server
-        working-directory: mcp-server
-        run: pnpm install --frozen-lockfile && pnpm run build
 
       - name: Build app
         working-directory: langwatch

--- a/.github/workflows/langwatch-app-ci.yml
+++ b/.github/workflows/langwatch-app-ci.yml
@@ -399,13 +399,13 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-turbo-
 
-      - name: Install dependencies
-        working-directory: langwatch
-        run: CI=true pnpm install --frozen-lockfile
-
       - name: Build mcp-server
         working-directory: mcp-server
         run: pnpm install --frozen-lockfile && pnpm run build
+
+      - name: Install dependencies
+        working-directory: langwatch
+        run: CI=true pnpm install --frozen-lockfile
 
       - name: Build
         working-directory: langwatch

--- a/.github/workflows/langwatch-app-ci.yml
+++ b/.github/workflows/langwatch-app-ci.yml
@@ -403,6 +403,10 @@ jobs:
         working-directory: langwatch
         run: CI=true pnpm install --frozen-lockfile
 
+      - name: Build mcp-server
+        working-directory: mcp-server
+        run: pnpm install --frozen-lockfile && pnpm run build
+
       - name: Build
         working-directory: langwatch
         run: pnpm build

--- a/.github/workflows/langwatch-server-publish.yml
+++ b/.github/workflows/langwatch-server-publish.yml
@@ -32,7 +32,7 @@ jobs:
 
       - name: Build mcp-server
         working-directory: mcp-server
-        run: pnpm install && pnpm run build
+        run: pnpm install --frozen-lockfile && pnpm run build
 
       - name: Install dependencies
         run: |

--- a/.github/workflows/langwatch-server-publish.yml
+++ b/.github/workflows/langwatch-server-publish.yml
@@ -30,6 +30,10 @@ jobs:
         with:
           version: 10.24.0
 
+      - name: Build mcp-server
+        working-directory: mcp-server
+        run: pnpm install && pnpm run build
+
       - name: Install dependencies
         run: |
           cd langwatch

--- a/.github/workflows/sdk-javascript-ci.yml
+++ b/.github/workflows/sdk-javascript-ci.yml
@@ -162,7 +162,9 @@ jobs:
         with:
           node-version: 24
           cache: "pnpm"
-          cache-dependency-path: "langwatch/pnpm-lock.yaml"
+          cache-dependency-path: |
+            langwatch/pnpm-lock.yaml
+            mcp-server/pnpm-lock.yaml
 
       - name: Install LangWatch app dependencies
         working-directory: langwatch
@@ -192,6 +194,10 @@ jobs:
       - name: Setup ClickHouse
         working-directory: langwatch
         run: pnpm clickhouse:migrate
+
+      - name: Build mcp-server
+        working-directory: mcp-server
+        run: pnpm install --frozen-lockfile && pnpm run build
 
       # Build the app so we can run a production server (dev mode is
       # unreliable in CI — turbopack JIT compilation can hang under

--- a/.github/workflows/sdk-javascript-ci.yml
+++ b/.github/workflows/sdk-javascript-ci.yml
@@ -166,6 +166,10 @@ jobs:
             langwatch/pnpm-lock.yaml
             mcp-server/pnpm-lock.yaml
 
+      - name: Build mcp-server
+        working-directory: mcp-server
+        run: pnpm install --frozen-lockfile && pnpm run build
+
       - name: Install LangWatch app dependencies
         working-directory: langwatch
         run: CI=true pnpm install --frozen-lockfile
@@ -194,10 +198,6 @@ jobs:
       - name: Setup ClickHouse
         working-directory: langwatch
         run: pnpm clickhouse:migrate
-
-      - name: Build mcp-server
-        working-directory: mcp-server
-        run: pnpm install --frozen-lockfile && pnpm run build
 
       # Build the app so we can run a production server (dev mode is
       # unreliable in CI — turbopack JIT compilation can hang under

